src/signature.rs

```diff
@@ -311,13 +311,17 @@ fn build_canonical_query_string(params: &Params) -> String {
     }
 
     let mut output = String::new();
-    for item in params.iter() {
+    for (key, val) in params.iter() {
         if !output.is_empty() {
             output.push_str("&");
         }
-        output.push_str(&encode_uri_strict(item.0));
-        output.push_str("=");
-        output.push_str(&encode_uri_strict(item.1));
+
+        output.push_str(&encode_uri_strict(key));
+        output.push_str("=");        
```

Comment:
Should this `=` be inside the `if` statement just below?  So a url can have `foo?bar` instead of `foo?bar=` if there's no value?